relay-dynamic-config/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+//!
+//! # Project Configuration
+//!
+//! So far, the only scope of dynamic configuration is per [`relay_auth::PublicKey`] a.k.a. [DSN](https://docs.sentry.io/product/sentry-basics/dsn-explainer/).
+//! The schema for this configuration is defined in [`ProjectConfig`].
+//!
+//! ## Example Config
+//!
+//! ```json
+//! {
+//!     "organizationId": 1,
+//!     "config": {
+//!         "excludeFields": [],
+//!         "filterSettings": {},
+//!         "scrubIpAddresses": False,
+//!         "sensitiveFields": [],
+//!         "scrubDefaults": True,
+//!         "scrubData": True,
+//!         "groupingConfig": {
+//!             "id": "legacy:2019-03-12",
+//!             "enhancements": "eJybzDhxY05qemJypZWRgaGlroGxrqHRBABbEwcC",
+//!         },
+//!         "blacklistedIps": ["127.43.33.22"],
+//!         "trustedRelays": [],
+//!     },
+//! }
+//! ```
+//!
+//!
+#![warn(missing_docs)]
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/getsentry/relay/master/artwork/relay-icon.png",
+    html_favicon_url = "https://raw.githubusercontent.com/getsentry/relay/master/artwork/relay-icon.png"
+)]
+#![allow(clippy::derive_partial_eq_without_eq)]
+
+mod error_boundary;
+pub use error_boundary::*;
+mod metrics;
+pub use metrics::*;
+use relay_quotas::Quota;
+
+use std::collections::BTreeSet;
+
+use relay_filter::FiltersConfig;
+use relay_general::pii::{DataScrubbingConfig, PiiConfig};
+use relay_general::store::{BreakdownsConfig, MeasurementsConfig, TransactionNameRule};
+use relay_general::types::SpanAttribute;
+use relay_sampling::SamplingConfig;
+use serde::{Deserialize, Serialize};
+
+use relay_auth::PublicKey;
+use serde_json::Value;
+
+/// Features exposed by project config.
+#[derive(Clone, Copy, Debug, Eq, PartialEq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+pub enum Feature {
+    /// Enables ingestion and normalization of profiles.
+    #[serde(rename = "organizations:profiling")]
+    Profiling,
+    /// Enables ingestion of Session Replays (Replay Recordings and Replay Events).
+    #[serde(rename = "organizations:session-replay")]
+    SessionReplay,
+    /// Enables data scrubbing of replay recording payloads.
+    #[serde(rename = "organizations:session-replay-recording-scrubbing")]
+    SessionReplayRecordingScrubbing,
+    /// Enables transaction names normalization.
+    ///
+    /// Replacing UUIDs, SHAs and numerical IDs by placeholders.
+    #[serde(rename = "organizations:transaction-name-normalize")]
+    TransactionNameNormalize,
+
+    /// Unused.
+    ///
+    /// This used to control the initial experimental metrics extraction for sessions and has been
+    /// discontinued.
+    #[serde(rename = "organizations:metrics-extraction")]
+    Deprecated1,
+
+    /// Forward compatibility.
+    #[serde(other)]
+    Unknown,
+}
+
+/// Dynamic, per-DSN configuration passed down from Sentry.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(default, rename_all = "camelCase")]
+pub struct ProjectConfig {
```

Comment:
Could we also move the `Feature` and `ProjectConfig` and basically all the code from here to its own module, like `config.rs` and just leave the `lib.rs` for docs and re-imports. 

In my opinion it would make the structure a little bit more cleaner.

Reply:
Done!